### DISCUSSION TITLE
Exoplayer transcoding issues?

### DISCUSSION THREAD

**Discussion by theinfguy** [jellyfin/jellyfin-androidtv#2983] [General]
Using AndroidTV Jellyfin client 0.15.11 on a Firestick 4K Max or Nvidia Shield TV Pro with Exoplayer selected, I am able to play certain files back direct stream just fine, however if I reduce the quality setting to force transcoding then I will get a player error. There are no client logs, and the server logs don't appear to show anything obvious.

I have attached a transcode and server log. You'll see theres no errors which leads me to believe this is a problem with Exoplayer in the AndroidTV client.

The logs show me starting playback in full direct stream of the movie Bad Boys II in 4K HVEC, HDR10 with DTS-MA audio, and then me hitting the quality button at 10Mbit/s to force a transcode.

If I change to LibVLC I am able to transcode these files just fine, however theres audio sync issues, the quality perhaps isn't as good and I am not sure on the future of libVLC as a supported backend?

[Server Log](https://www.dropbox.com/scl/fi/i2p66wc4wq0rf1zuhyioi/Server-Log.txt?rlkey=4qbohfcdj0pas8fothkad3t0w&dl=0)

[Transcode Log](https://www.dropbox.com/scl/fi/wng7v2iyz1uaqbofn4rzs/Transcode-log.txt?rlkey=zyqy2gnlo69p0i0vu5tm3d780&dl=0)

Mediainfo of files which WON'T transcode in Exo:

`Bad Boys II_2003_4K
Container: mkv
Path: /mnt/bigstor/Tom Dropbox/Media/Movies/B/Bad Boys Collection/Bad Boys II_2003_4K.mkv
Size: 26009 MB

Video
Title: 4K HEVC HDR
Codec: HEVC
Profile: Main 10
Level: 150
Resolution: 3840x1606
Aspect ratio: 2.40:1
Interlaced: No
Framerate: 23.976025
Bitrate: 24777 kbps
Bit depth: 10 bit
Video range: HDR
Video range type: HDR10
Colour space: bt2020nc
Colour transfer: smpte2084
Colour primaries: bt2020
Pixel format: yuv420p10le
Ref frames: 1

Audio
Title: English - DTS-HD MA - 7.1 - Default
Language: eng
Codec: DTS
Profile: DTS-HD MA
Layout: 7.1
Channels: 8 ch
Sample rate: 48000 Hz
Bit depth: 24 bit
Default: Yes
Forced: No
External: No

Subtitle
Title: English - PGSSUB
Language: eng
Codec: PGSSUB
Default: No
Forced: No
External: No

-------------------

American Gangster_2007_4K
Container: mkv
Path: /mnt/bigstor/Tom Dropbox/Media/Movies/A/American Gangster_2007_4K.mkv
Size: 17923 MB

Video
Title: 4K HEVC HDR
Codec: HEVC
Profile: Main 10
Level: 153
Resolution: 3840x2076
Aspect ratio: 1.85:1
Interlaced: No
Framerate: 23.976025
Bitrate: 14254 kbps
Bit depth: 10 bit
Video range: HDR
Video range type: HDR10
Colour space: bt2020nc
Colour transfer: smpte2084
Colour primaries: bt2020
Pixel format: yuv420p10le
Ref frames: 1

Audio
Title: English DTS-X 7.1 - DTS-HD MA - Default
Language: eng
Codec: DTS
Profile: DTS-HD MA
Layout: 7.1
Channels: 8 ch
Sample rate: 48000 Hz
Bit depth: 24 bit
Default: Yes
Forced: No
External: No

Subtitle
Title: English SRT - Default - SUBRIP
Language: eng
Codec: SUBRIP
Default: Yes
Forced: No
External: No

Audio
Title: English DD 5.1 - Dolby Digital
Language: eng
Codec: AC3
Layout: 5.1
Channels: 6 ch
Bitrate: 384 kbps
Sample rate: 48000 Hz
Default: No
Forced: No
External: No

Subtitle
Title: English SDH SRT - SUBRIP
Language: eng
Codec: SUBRIP
Default: No
Forced: No
External: No

Image
Codec: MJPEG
Profile: Baseline
Resolution: 550x825
Bit depth: 8 bit
Colour space: bt470bg
Pixel format: yuvj420p
Ref frames: 1`

These not the only files this happens to, but it doesn't happen on everything. I am wondering if someone can help? Again, this only happens when using Exoplayer in the AndroidTV client, switching to LibVLC works fine!

Also have a post on the forum:

https://forum.jellyfin.org/t-androidtv-exoplayer-unable-to-transcode-certain-files

**Comment by averagejooe** [comment]
Off topic answer but really why would you transcode 4K? Its very resource intensive and will never yield a good results.

**Reply by theinfguy** [reply]
Erm, because you want a 4K source for when you have the bandwidth, and transcoding for when you don't. The Jellyfin server is on powerful hardware with intel hardware transcoding enabled. The server doesn't even blink.
